lib/config/flat-config-schema.js

```diff
@@ -438,11 +454,40 @@ const sourceTypeSchema = {
     }
 };
 
+/**
+ * Creates a schema that always throws an error. Useful for warning
+ * about eslintrc-style keys.
+ * @param {string} key The eslintrc key to create a schema for.
+ * @returns {ObjectPropertySchema} The schema.
+ */
+function createEslintrcSchema(key) {
+    return {
+        merge: "replace",
```

Comment:
I don't think `config-array`/`object-schema` require `merge` to be present, and since a config that this schema applies to won't ever be merged because it's invalid, can `merge` be removed?